src/lib.rs

```diff
@@ -0,0 +1,420 @@
+            let blog_owner = ensure_signed(origin)?;
+            let blogs_count = Self::blogs_count();
+            if <BlogIds<T>>::exists(&blog_owner) {
+                <BlogIds<T>>::mutate(&blog_owner, |blog_ids| {
+                    if let Some(blog_ids) = blog_ids {
+                        blog_ids.insert(blogs_count);
+                    }
+                })
+            } else {
+                let mut new_set = BTreeSet::new();
+                new_set.insert(blogs_count);
+                <BlogIds<T>>::insert(&blog_owner, new_set);
+            }
+            // Blog locking default status
+            <BlogLockedStatus<T>>::insert(blogs_count, false);
+            <BlogsCount<T>>::mutate(|count| *count += T::BlogId::one());
+            Self::deposit_event(RawEvent::BlogCreated(blog_owner, blogs_count));
+            Ok(())
+        }
+
+        pub fn lock_blog(origin, blog_id: T::BlogId) -> dispatch::Result {
+            let blog_owner = ensure_signed(origin)?;
+            ensure!(<BlogIds<T>>::exists(&blog_owner), BLOG_OWNER_NOT_FOUND);
+            match Self::blog_ids_by_owner(&blog_owner) {
+                Some(blog_ids_set) if blog_ids_set.contains(&blog_id) => {
+                    <BlogLockedStatus<T>>::mutate(&blog_id, |locked_status| *locked_status = true)
+                }
+                _ => return Err(BLOG_OWNERSHIP_ERROR)
+            }
+            Self::deposit_event(RawEvent::BlogLocked(blog_owner, blog_id));
+            Ok(())
+        }
+
+        pub fn unlock_blog(origin, blog_id: T::BlogId) -> dispatch::Result {
+            let blog_owner = ensure_signed(origin)?;
+            ensure!(<BlogIds<T>>::exists(&blog_owner), BLOG_OWNER_NOT_FOUND);
+            match Self::blog_ids_by_owner(&blog_owner) {
+                Some(blog_ids_set) if blog_ids_set.contains(&blog_id) => {
+                    <BlogLockedStatus<T>>::mutate(&blog_id, |locked_status| *locked_status = false)
+                }
+                _ => return Err(BLOG_OWNERSHIP_ERROR)
+            }
+            Self::deposit_event(RawEvent::BlogUnlocked(blog_owner, blog_id));
+            Ok(())
+        }
+
+        pub fn create_post(origin, blog_id: T::BlogId, title: Vec<u8>, body: Vec<u8>) -> dispatch::Result  {
+            let blog_owner = ensure_signed(origin)?;
+            ensure!(<BlogIds<T>>::exists(&blog_owner), BLOG_OWNER_NOT_FOUND);
+            ensure!(!Self::blog_locked(blog_id), BLOG_LOCKED_ERROR);
+            // Check security/configuration constraints
+            ensure!(title.len() <= T::PostTitleMaxLength::get() as usize, POST_TITLE_TOO_LONG);
+            ensure!(body.len() <= T::PostBodyMaxLength::get() as usize, POST_BODY_TOO_LONG);
+            if let Some(post_ids) = Self::post_ids_by_blog_id(blog_id) {
+                ensure!(post_ids.len() < T::PostsMaxNumber::get() as usize, POSTS_LIMIT_REACHED)
+            }
+            let posts_count = Self::posts_count(blog_id);
```

Comment:
This code flow is a great example of what I was describing above, we must try to avoid it.

Reply:
Agree

Reply:
https://github.com/Joystream/substrate-blog-module/pull/1/commits/a682cc3630bf28aa110ac0c0186936885c3fc2e3#diff-b4aea3e418ccdb71239b96952d9cddb6R294